src/qt/receivecoinsdialog.cpp

```diff
@@ -84,6 +84,12 @@ void ReceiveCoinsDialog::setModel(WalletModel* model)
             SLOT(recentRequestsView_selectionChanged(QItemSelection, QItemSelection)));
         // Last 2 columns are set by the columnResizingFixer, when the table geometry is ready.
         columnResizingFixer = new GUIUtil::TableViewLastColumnResizingFixer(tableView, AMOUNT_MINIMUM_COLUMN_WIDTH, DATE_COLUMN_WIDTH);
+
+        // Init address field
+        address = getAddress();
```

Comment:
Last unused address should be made persistent and reloaded when the wallet starts.
Otherwise a new one is created each time (whether it has been used or not).